app/email_processors/email_processor.rb

```diff
@@ -0,0 +1,37 @@
+class EmailProcessor
+
+  def self.process(*args)
+    new(*args).process
+  end
+
+  def initialize(email)
+    @email = replace_text_body(email)
+  end
+
+
+  def process
+    email_params = EmailParams.new(@email)
+    user = User.find_by(id: email_params.user_id,
+                        email_api_key: email_params.email_api_key)
+
+    comment = Comment.new(discussion_id: email_params.discussion_id,
+                          body: email_params.body)
+
+    CommentService.create(comment: comment, actor: user)
```

Comment:
Trying to wrap my head around what happens when, say the user isn't found via their API key. Does it fail gracefully :grey_question: